pre-merge update: rasp-pi tutorial (#6633)
ethereum/ethereum-org-website: src/content/developers/tutorials/run-node-raspberry-pi/index.md

---
title: How to turn your Raspberry Pi 4 into a node just by flashing the MicroSD card
description: Flash your Raspberry Pi 4, plug in an ethernet cable, connect the SSD disk and power up the device to turn the Raspberry Pi 4 into a full Ethereum node + validator
author: "EthereumOnArm"
tags: ["clients", "execution layer", "consensus layer", "nodes"]
lang: en
sidebar: true
skill: intermediate
published: 2022-06-10
source: [Ethereum on ARM](https://ethereum-on-arm-documentation.readthedocs.io/en/latest/kiln/kiln-testnet.html)
sourceUrl: https://www.reddit.com/r/ethereum/comments/gf3nhg/ethereum_on_arm_raspberry_pi_4_images_release/
---

**Ethereum on Arm is a custom Linux image that can turn a Raspberry Pi into an Ethereum node.**

To use Ethereum on Arm to turn a Raspberry Pi into an Ethereum node, the following hardware is recommended:

- Raspberry 4 (model B 8GB)
- MicroSD Card (16 GB Class 10 minimum)

> - 1 TB SSD minimum USB 3.0 disk or an SSD with an USB to SATA case (see Storage section).

Reviewer: Would recommend 2 TB min right now unfortunately... Geth already runs ~1 TB alone.
Also, the `#storage` header was removed, so there is no "Storage" section =)

```suggestion
- 2 TB SSD minimum USB 3.0 disk or an SSD with a USB to SATA case.
```

Reviewer 2: +1

Once everything is installed and configured, log in to the device via an ssh connection or using the terminal directly if a monitor and keyboard is attached to the board. Use the `ethereum` account to log in, as this has permissions required to start the node.

```shell
User: ethereum
Password: ethereum
```

The user can then choose the execution-consensus client combination they wish to run, and start their systemctl processes as follows (example runs Geth and Lighthouse):

```shell
sudo systemctl start geth-lh
sudo systemctl start lh-geth-beacon
```

The logs can be checked using

```shell
# logs for Geth
sudo journalctl -u geth-lh -f
#logs for lighthouse
sudo journalctl -u lh-geth-beacon -f
```

The specific service names for every combination of clients are available at the [Ethereum on Arm docs](https://ethereum-on-arm-documentation.readthedocs.io/en/latest/kiln/kiln-testnet.html#id2). Theyu can be used to update the commands provided here for any combination.

## Validators {#validators}

> In order to run a validator it is first necessary to deposit 32 test-ether into the Kiln deposit contract. This means the user must have access to an account containing 32 ETH. This can be done by following the step-by-step guide on the [Kiln Launchpad](https://kiln.launchpad.ethereum.org/en/). Do this on a desktop/laptop, built do not generate keys - this can be done directly on the Raspberry Pi.

Reviewer: > This means the user must have access to an account containing 32 ETH. This can be done by following the step-by-step guide on the [Kiln Launchpad](https://kiln.launchpad.ethereum.org/en/)

This kinda sounds like there is a faucet on the Launchpad. 

```suggestion
In order to run a validator you must first have access to 32 testnet ETH, which must be deposited into the Kiln deposit contract. This can be done by following the step-by-step guide on the [Kiln Launchpad](https://kiln.launchpad.ethereum.org/en/). Do this on a desktop/laptop, but do not generate keys—this can be done directly on the Raspberry Pi.
```

## How does it work? {#how-does-it-work}

The Raspberry Pi's memory card is flashed with a prebuilt image. This image contains everything needed to run an Ethereum node. With a flashed card, all the user needs to do is power-on the Raspberry Pi. All the processes required to run the node are automatically started. This works because the memory card contains a Linux-based operating system (OS) on top of which system-level processes are automatically run that turn the unit into an Ethereum node.

Ethereum cannot be run using the popular Raspberry Pi Linux OS "Raspbian" because Raspbian still uses a 32-bit architecture which leads Ethereum users to run into memory issues and consensus clients do not support 32-bit binaries. To overcome this, the Ethereum on Arm team migrated to a native 64-bit OS called "Armbian".

**Images take care of all the necessary steps**, from setting up the environment and formatting the SSD disk to installing and running the Ethereum software as well as starting the blockchain synchronization.

## Note on execution and consensus clients {#note-on-execution-and-consensus-clients}

> The Ethereum on Arm documentation explains how to set up _either_ an execution client OR a consensus client, except for two Ethereum testnets (Kiln and Ropsten). This optionality is only possible in advance of Ethereum's upcoming transition from proof-of-work (PoW) to proof-of-stake (PoS) known as [The Merge](/upgrades/merge). After The Merge, it will not be possible to run execution and consensus clients separately - they must be run as a pair. Therefore, in this tutorial we will run a pair of execution and consensus clients together on an Ethereum testnet (Kiln).

Reviewer: This is great! Can I suggest we emphasize this section a little more given it has some time-sensitive information? We can use the `InfoBanner` component (already available for this template), perhaps on the second half of this paragraph:

```suggestion
The Ethereum on Arm documentation explains how to set up _either_ an execution client OR a consensus client, except for two Ethereum testnets (Kiln and Ropsten). This optionality is only possible in advance of Ethereum's upcoming transition from proof-of-work (PoW) to proof-of-stake (PoS) known as [The Merge](/upgrades/merge).

<InfoBanner>
After The Merge, it will not be possible to run execution and consensus clients separately—they must be run as a pair. Therefore, in this tutorial we will run a pair of execution and consensus clients together on an Ethereum testnet (Kiln).
</InfoBanner>
```

Reviewer 2: +1

Kiln is a public testnet specifically designed for testing The Merge. Ethereum on Arm developed an image allowing users to rapidly spin up a pair of Ethereum clients on this merge testnet. The Kiln merge has already happened, but the network is still live, so it can be used for this tutorial. Ether on Kiln has no real-world value.

The Kiln Raspberry Pi 4 image is a "plug and play" image that automatically installs and sets up both the execution and consensus clients, configuring them to talk to each other and connect to the Kiln network. All the user needs to do is start their processes using a simple command. The image contains four execution clients (Geth, Nethermind, Besu and Erigon) and four consensus clients (Lighthouse, Prysm, Nimbus, Teku) that the user can choose from.

Download the Raspberry Pi image from [Ethereum on Arm](https://ethereumonarm-my.sharepoint.com/:u:/p/dlosada/ES56R_SuvaVFkiMO1Tgnf6kB7lEbBfla5c2c18E3WQRJzA?download=1) and verify the SHA256 hash:

```sh
# From directory containing the downloaded image
shasum -a 256 ethonarm_kiln_22.03.01.img.zip
# Hash should output: 485cf36128ca60a41b5de82b5fee3ee46b7c479d0fc5dfa5b9341764414c4c57

> ```

Reviewer: (Added in a code snippet here so users can easily verify the sha256 hash, in case they didn't know how to do it)
